### DISCUSSION TITLE
Run nuclei with specific template id

### DISCUSSION THREAD

**Discussion by CaledoniaProject** [projectdiscovery/nuclei#1302] [Q&A] [answered]
I'm trying to run specific template id, but this does not work:

```
nuclei -u https://XX.com -w ~/nuclei-templates/miscellaneous/dir-listing.yaml
nuclei -u https://XX.com -w dir-listing
```

Error is:

```
[WRN] Use with caution. You are responsible for your actions.
[WRN] Developers assume no liability and are not responsible for any misuse or damage.
[INF] Using Nuclei Engine 2.5.3 (latest)
[INF] Using Nuclei Templates 8.6.8 (latest)
[INF] Using Interactsh Server https://interactsh.com
[FTL] Could not run nuclei: no valid templates were found
```

**Comment by ehsandeep** [comment, marked as answer]
@CaledoniaProject `-w` flag is used for workflows, for templates, you need to use `-t` flag and templates can be executed either by passing template file or template directory.

```
nuclei -t miscellaneous/dir-listing.yaml -itags misc
```

**Comment by CaledoniaProject** [comment]
@ehsandeep I need to avoid the `-itags misc` argument, I just want to run a yaml file. What options do I have?

From what I see, `-itags` is only useful when I'm selecting multiple yaml files in a large directory. In my case I just need one specific file.

**Comment by infosec-us-team** [comment]
I can confirm I'm running into the same issue.

This is the command and the output:
```
infusero@RemoteWorkstation:~/reconnaissance-everything$ ~/go/bin/nuclei -t tools/nuclei/custom-template/detect-graphql.yaml -u https://example.com -v

                     __     _
   ____  __  _______/ /__  (_)
  / __ \/ / / / ___/ / _ \/ /
 / / / / /_/ / /__/ /  __/ /
/_/ /_/\__,_/\___/_/\___/_/   2.5.3

                projectdiscovery.io

[WRN] Use with caution. You are responsible for your actions.
[WRN] Developers assume no liability and are not responsible for any misuse or damage.
[WRN] Could not load template /home/infusero/reconnaissance-everything/tools/nuclei/custom-template/detect-graphql.yaml: the template was excluded
[INF] Using Nuclei Engine 2.5.3 (latest)
[INF] Using Nuclei Templates 8.6.8 (latest)
[INF] Using Interactsh Server https://interactsh.com
[FTL] Could not run nuclei: no valid templates were found
```

**Comment by ehsandeep** [comment]
@CaledoniaProject @infosec-us-team `itags / include-tags` are only required when you wanted to run templates that are excluded from the default run, read more about it here - https://nuclei.projectdiscovery.io/nuclei/get-started/#template-exclusion, you can run any other template without `itags` flag, for example `-t cves/`

**Comment by infosec-us-team** [comment]
As I'm new to Nuclei (started today), plus I'm stup1d, I just removed the `fuzz` tag from my custom-made template because by default nuclei exclude templates with `fuzz`.

Now is working fine.

Thank you for your fast response @ehsandeep, you are awesome.
